src/App.tsx

```diff
@@ -1,27 +1,53 @@
 /* eslint-disable max-len */
-/* eslint-disable no-console */
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { getTodos } from './api';
-import { Todo } from './types/Todo';
-import { Loader } from './components/Loader';
 import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { getTodos /* getUser */ } from './api';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
   const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [modal, setModal] = useState(false);
+  const [filter, setFilter] = useState<Filter>('all');
+  const [searchTerm, setSearchTerm] = useState('');
   const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+
+  const visibleTodos = () => {
+    const visible = [...todos].filter((todo) => {
+      if (filter === 'active' && todo.completed) {
+        return false;
+      }
+
+      if (filter === 'completed' && !todo.completed) {
+        return false;
+      }
+
+      if (!todo.title.toLowerCase().includes(searchTerm.toLowerCase())) {
+        return false;
+      }
+
+      return true;
+    });
```

Comment:
that would be a perfect usecase for switch - we're having finite number of cases here

Reply:
Like that:
const visibleTodos = () => {
  const visible = [...todos].filter((todo) => {
    switch (filter) {
      case 'active':
        if (todo.completed) return false;
        break;

      case 'completed':
        if (!todo.completed) return false;
        break;
    }

    if (!todo.title.toLowerCase().includes(searchTerm.toLowerCase())) {
      return false;
    }

    return true;
  });

  return visible;
};

Reply:
Can this be done better? Because in this form, it's less readable.

Reply:
It might seem that way, but I think it separates the filtering logic from a search phrase and if you were to add another filter type you would know where to put it right away. And when it comes to doing it better - I think that you could possibly do a loop through all the cases but I am not sure if it would be as readable as it is right now

Reply:
Well, then maybe let's leave it at the way it is ;D